flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/catalog/AbstractJdbcCatalog.java

```diff
@@ -116,6 +122,36 @@ public String getBaseUrl() {
 		return baseUrl;
 	}
 
+	// ------ Postgres default objects that shouldn't be exposed to users ------
+
+	/**
+	 * Retrieve the list of system schemas to ignore.
+	 */
+	public abstract Set<String> getBuiltinSchemas();
+
+	/**
+	 * Retrieve the list of system database to ignore.
+	 */
+	public abstract Set<String> getBuiltinDatabases();
+
+	// ------ retrieve PK constraint ------
+
+	protected Map.Entry<String, List<String>> getPrimaryKey(DatabaseMetaData metaData, PostgresTablePath pgPath) throws SQLException {
```

Comment:
If we want to move this method into the base class, I would suggest to change the signature into 

```java
UniqueConstraint getPrimaryKey(DatabaseMetaData metaData, String schema, String table)
```

1. `UniqueConstraint` is a standard public API to describe primary key which is exposed by Table API.
2. `PostgresTablePath` is a postgres speicifc class which shouldn't be in the base class.